contracts/GenesisSupply.sol

```diff
@@ -226,15 +249,20 @@ contract GenesisSupply is AccessControl {
         validTokenId(tokenId)
         returns (TokenTraits memory traits)
     {
+        // Backend has access to metadata before anyone
+        if (hasRole(BACKEND_ROLE, msg.sender)) {
+            return tokenIdToTraits[tokenId];
+        }
         if (isRevealed) {
```

Comment:
Micro-optimizations, I would put `isRevealed` condition first so most call cost barely nothing to run, only when not reveal that extra gas will need to be paid.

Reply:
Can't do that because backend has access to this BEFORE the reveal date to be able to generate the NFTs before the reveal.